contrib/database/sql/sql.go

```diff
@@ -12,49 +12,169 @@
 package sql
 
 import (
+	"context"
 	"database/sql"
 	"database/sql/driver"
 	"errors"
+	"math"
+	"reflect"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/contrib/database/sql/internal"
 )
 
+// registeredDrivers holds a registry of all drivers registered via the sqltrace package.
+var registeredDrivers = &driverRegistry{
+	keys:    make(map[reflect.Type]string),
+	drivers: make(map[string]driver.Driver),
+	configs: make(map[string]*config),
+}
+
+type driverRegistry struct {
+	// keys maps driver types to their registered names.
+	keys map[reflect.Type]string
+	// drivers maps keys to their registered driver.
+	drivers map[string]driver.Driver
+	// configs maps keys to their registered configuration.
+	configs map[string]*config
+}
+
+// registered checks if the named driver has already been registered.
```

Comment:
A practice I've been perpetuating is to use the word "_reports_" when boolean values are returned. I've seen this used throughout the Go standard library too and I think it's a good way to indicate what a return value of `true` means. e.g.
```go
// isRegistered reports whether the driver having name is in the registry.
```
You may, of course, use better wording :)

Reply:
I used similar wording, but especially the `reports whether`. I hadn't noticed it before, but yup - it's there a lot and is clearer than `checks if`.